common/datasets/switchboard/eval.py

```diff
@@ -0,0 +1,132 @@
+"""
+All functions and classes related to switchboard evaluation sets
+"""
+__all__ = ["SwitchboardEvalDataset", "get_hub5e00", "get_hub5e01", "get_rt03s"]
+from functools import lru_cache
+from dataclasses import dataclass
+from sisyphus import tk
+
+from i6_core.meta.system import CorpusObject
+from i6_core.datasets.switchboard import (
+    SwitchboardSphereToWaveJob,
+    CreateHub5e00CorpusJob,
+    CreateHub5e01CorpusJob,
+    CreateRT03sCTSCorpusJob,
+)
+from i6_experiments.common.datasets.switchboard.constants import durations
+from i6_experiments.common.datasets.switchboard.paths import (
+    HUB5E00_SPH_PATH,
+    HUB5E00_TRANSCRIPT_PATH,
+    HUB5E01_PATH,
+    RT03S_PATH,
+)
+
+
+@dataclass(frozen=True)
+class SwitchboardEvalDataset:
+    """
+    A dataclass helper to unify the objects belonging to a single evaluation set from Switchboard
+    """
+
+    bliss_corpus: tk.Path
+    stm: tk.Path
+    glm: tk.Path
+
+
+@lru_cache()
+def get_hub5e00() -> SwitchboardEvalDataset:
+    """
+    :return: hub5e00 eval dataset
+    """
+    hub5e00_wav_audio = SwitchboardSphereToWaveJob(
+        sph_audio_folder=HUB5E00_SPH_PATH
+    ).out_wave_audio_folder
+    hub5e00_job = CreateHub5e00CorpusJob(
+        wav_audio_folder=hub5e00_wav_audio,
+        hub5_transcription_folder=HUB5E00_TRANSCRIPT_PATH,
+    )
+    return SwitchboardEvalDataset(
+        bliss_corpus=hub5e00_job.out_bliss_corpus,
+        stm=hub5e00_job.out_stm,
+        glm=hub5e00_job.out_glm,
+    )
+
+
+@lru_cache()
+def get_hub5e00_corpus_object() -> CorpusObject:
+    """
+    :return: hub5e00 corpus object
+    """
+    hub5e00 = get_hub5e00()
+    hub5e00_corpus_object = CorpusObject()
+    hub5e00_corpus_object.corpus_file = hub5e00.bliss_corpus
+    hub5e00_corpus_object.audio_format = "wav"
+    hub5e00_corpus_object.audio_dir = None
+    hub5e00_corpus_object.duration = durations["hub5e00"]
+    return hub5e00_corpus_object
```

Comment:
In my opinion it would be better to return a copy of the object here. Otherwise its easy for the user to make an error and overwrite something in the corpus_object and manipulate more experiments than desired.

Reply:
solved via frozen dataclass